ktor-server/ktor-server-netty/jvm/src/io/ktor/server/netty/http2/Http2LocalConnectionPoint.kt

```diff
@@ -25,12 +25,13 @@ internal class Http2LocalConnectionPoint(
         get() = nettyHeaders.path()?.toString() ?: "/"
 
     override val host: String
-        get() = nettyHeaders.authority()?.toString() ?: "localhost"
+        get() = nettyHeaders.authority()?.toString()?.substringBefore(":") ?: "localhost"
 
     override val port: Int
-        get() = nettyHeaders.authority()?.toString()?.substringAfter(":")?.toInt()
-                ?: localAddress?.port
-                ?: 80
+        get() = nettyHeaders.authority()?.toString()
+            ?.substringAfter(":", "-1")?.toInt()?.takeIf { it > 0 }
```

Comment:
I would prefer not to use `-1` as a fallback, like `?.substringAfter(":", "")?.takeIf { it.isNotEmpty() }?.toInt()`, but it's up to you